File: 392. Is Subsequence/392.js
// 392. Is Subsequence solution in JavaScript
// https://leetcode.com/problems/is-subsequence/description/

/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
    let s_len = s.length;
    let t_len = t.length;

    let i = 0;
    let j = 0;

    while (i < s_len && j < t_len) {
        if (s[i] == t[j]) {
            i++;
            j++;
        } else {
            j++;
        }
    }

    if (i == s_len) return true;

    return false;
};

File: 392. Is Subsequence/392.java
// 392. Is Subsequence solution in Java
// https://leetcode.com/problems/is-subsequence/description/

class Solution {
    public boolean isSubsequence(String s, String t) {
        int len_s = s.length();
        int len_t = t.length();
        int pointer_s = 0;
        int pointer_t = 0;

        while (pointer_s < len_s && pointer_t < len_t) {
            if (s.charAt(pointer_s) == t.charAt(pointer_t)) {
                pointer_s++;
            }
            pointer_t++;
        }
        
        return len_s == pointer_s;
    }
}